src/remote.js

```diff
@@ -194,64 +202,75 @@
-                      var buildDir = "";
-                      shell.mkdir('-p', path.join(cwd, 'build'));
-                      monaca.download(sessionUrl, {}, function(response) {
-                        var filename = "";
-                        if (typeof response.headers['content-disposition'] === 'string') {
-                          filename = response.headers['content-disposition'].match(/filename="?([^"]+)"?/)[1];
-                        }
-                        buildDir = path.join(cwd, "build", filename);
-                        return buildDir;
-                      }
-                      ).then(
-                        function(name) {
-                          util.print("Your package is stored at " + buildDir);
+            monaca.checkBuildAvailability(projectInfo.projectId, params.platform, params.purpose)
+            .then(
+              function() {
+                // Open the browser if no platform parameter is provided.
+                if (!argv.platform) {
+                  var url = 'https://ide.monaca.mobi/project/' + projectInfo.projectId + '/' + (argv['build-type'] ? 'debugger' : 'build');
+                  monaca.getSessionUrl(url)
+                  .then(
+                    function(url) {
+                      open(url);
+                    },
+                    function(error) {
+                      util.err('Unable to open build page.');
+                      process.exit(1);
+                    }
+                  );
+                } else {
+                  // Build project on Monaca Cloud and download it into ./build folder.
+                  util.print("Building project on Monaca Cloud...");
+                  monaca.buildProject(projectInfo.projectId, params)
+                  .then(function(result) {
+                    if (result.binary_url) {
+                      return result.binary_url;
+                    } else {
+                      return Q.reject(result.error_message);
+                    }
+                  },
+                  function(err) {
+                    return Q.reject(err);
+                  },
+                  function(progress) {
+                    process.stdout.write(".");
+                  })
+                  .then(
+                    function(url) {
+                      monaca.getSessionUrl(url)
+                      .then(
+                        function(sessionUrl) {
+                          var buildPath = "";
+                          monaca.download(sessionUrl, {}, function(response) {
+                            var filename = "";
+                            if (params.output) {
+                              buildPath = params.output;
+                            } else {
+                              if (typeof response.headers['content-disposition'] === 'string') {
+                                filename = response.headers['content-disposition'].match(/filename="?([^"]+)"?/)[1];
```

Comment:
You should also confirm `content-disposition` header matches the regular expression before picking up the matching string.